### PR TITLE
Complete html-forms exercise

### PR DESCRIPTION
**Pull request by BenGrimmius**
[Screen Recording 2022-08-25 at 9.35.14 PM.mov.zip](https://github.com/BenGrimmius/mod1-0822-code-solutions/files/9429843/Screen.Recording.2022-08-25.at.9.35.14.PM.mov.zip)

I know I used a lot of divs and ps in this so it looks a a little messy. But i fixed all the lint errors.

### CODE REVIEW

**Review by scrumptiousdev — CHANGES_REQUESTED** [review]
Haha, it sure does have many `<p>`s! we would like to practice formatting the layout without using it though. I can definitely let this one go but since we are heading into the weekend, let's go ahead and practice!

Whenever you have a "group of form control" like `label` and `input`, we would wrap it around with `<div>` like below:
```
<div> // we wrap with div here to break line between this form control group and others
  <label for="something">Something</label>
  <div> // we wrap with div here to break line between label and input
    <input name="something" type="text" id="something">
  </div>
</div>
```

Hope this helps in order to clean things up!

**Comment by BenGrimmius** [comment]
For some reason when i did that it smashes everything together still🤔

**Comment by scrumptiousdev** [comment]
Yes indeed. But good job on implementing that in order to break the line between form controls! So now what we want to do is.. if you look carefully at our example web page gif, you will notice that we have two things that leaves a margin on top and bottom — `Favorite Language (choose one)` and `Would you like to sign up for more information?`. Those two can be a `<p>` because in this case, it's not a label to anything. Radio buttons have their own labels for each. And checkbox should have its own label attach to it with `Sign me up!` instead of `Would you like...`

Does this make sense?

File: html-forms/index.html [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        <select name="reason" id="Why">
+          <option value="select">Select an option</option>
+          <option value="interesting">Because it's interesting</option>
+          <option value="fun">Because it's fun</option>
+          <option value="powerful">Because it's powerful</option>
+        </select>
+      </div>
+    </div>
+
+    <div>Favorite Language</div>
```
Comment (scrumptiousdev) [inline html-forms/index.html]: This can be `<p>` since it's not a label

File: html-forms/index.html [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      <input type="radio" name="favorite" id="css" value="css">
+      <label for="css">CSS</label>
+      <input type="radio" name="favorite" id="javascript" value="Javascript">
+      <label for="javascript">Javascript</label>
+    </div>
+
+    <div>
+      <label for="signup">Would you like to signup for more information?</label>
+      <div>
+        <input type="checkbox" id="signup" name="signup" value="signup">
```
Comment (scrumptiousdev) [inline html-forms/index.html:37]: I think we are missing `Sign me up!` label for the checkbox :)
Then the `Would you like to signup...` can become a `<p>`

**Review by scrumptiousdev — CHANGES_REQUESTED** [review]

**Comment by BenGrimmius** [comment]
Oh im sorry ill add that last bit now

**Comment by BenGrimmius** [comment]
Screen Shot 2022-08-26 at 11 04 09 AM

File: html-forms/index.html [hunk 3]
```diff
@@ -0,0 +1,52 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <title>HTML Forms</title>
+</head>
+<body>
+  <h1>Coding Form!</h1>
+  <form>
+    <label for="Email">Email</label>
+    <input type="text" id="Email">
```
Comment (scrumptiousdev) [inline html-forms/index.html]: One last tiny thing we are missing here. We are missing the most important attribute — can you find what that is? And we are good to go!

**Review by scrumptiousdev — CHANGES_REQUESTED** [review]

File: html-forms/index.html (hunk 3 above)
Comment (BenGrimmius) [reply html-forms/index.html]: the name attr?

Comment (BenGrimmius) [reply html-forms/index.html]: and the value attr.

**Comment by BenGrimmius** [comment]
I added a name and value attribute to the input. tried to do it on the label but lint was giving me an error saying its an invalid attr

File: html-forms/index.html (hunk 3 above)
Comment (scrumptiousdev) [reply html-forms/index.html]: Just a `name` was a 🔑 ! Good job

**Review by scrumptiousdev — APPROVED** [review]
🎸 Approved!

**Pull request merged**